Framework/Algorithms/src/SolidAngle.cpp

```diff
@@ -26,6 +28,73 @@ DECLARE_ALGORITHM(SolidAngle)
 
 using namespace Kernel;
 using namespace API;
+using namespace Geometry;
+
+namespace {
+
+/**
+ * Returns the angle between the sample-to-pixel vector and its
+ * projection on the X-Z (vertical tube) or Y-Z (horizontal tube) plane.
+ * Note, in all cases Y is assumed to be the pointing-up direction, Z is the
+ * beam direction.
+ */
+double getTubeAngle(const DetectorInfo &detectorInfo, size_t index,
+                    const bool vertical) {
+  const auto sampleDetVec =
+      detectorInfo.position(index) - detectorInfo.samplePosition();
+  auto inPlane = sampleDetVec;
+  if (vertical)
```

Comment:
Might be worth using two separate functions since the value of `vertical` is known and remains constant the entire time.

Reply:
That leads to a duplicated code. Template method pattern might be the best option here.